### PR TITLE
Add :only option to deriving

### PR DESCRIPTION
**Pull request by HidetakaKojo**
Hi, Aleksei.

This PR add the `:only` option to the `@derive` clause in Msgpax.Packer.
The reason behind this addition is using with Ecto.

https://github.com/devinus/poison/pull/40#issuecomment-131323366

### CODE REVIEW

File: lib/msgpax/packer.ex [hunk 1]
```diff
@@ -207,15 +207,21 @@ defimpl Msgpax.Packer, for: Msgpax.Ext do
...
   end
 
-  def deriving(module) do
+  def deriving(module, opts) do
+    extractor = cond do
+      opts[:only] ->
+        quote(do: Map.take(struct, unquote(opts[:only])))
+      true ->
+        quote(do: Map.from_struct(struct))
+    end
```
Comment (whatyouhide) [inline lib/msgpax/packer.ex]: You can rewrite this as:

```elixir
extractor =
  if only = opts[:only] do
    quote(do: Map.take(struct, unquote(only)))
  else
    quote(do: Map.from_struct(struct))
  end
```

File: test/test_helper.exs [hunk 2]
```diff
@@ -25,9 +25,9 @@ defmodule Msgpax.Case do
 
   defp round_trip(input, format, output, opts) do
     quote do
-      assert {:ok, packed} = Msgpax.pack(unquote(input))
-      assert <<unquote_splicing(format), _::bytes>> = IO.iodata_to_binary(packed)
-      assert {:ok, unpacked} = Msgpax.unpack(packed, unquote(opts))
+      {:ok, packed} = Msgpax.pack(unquote(input))
+      <<unquote_splicing(format), _::bytes>> = IO.iodata_to_binary(packed)
+      {:ok, unpacked} = Msgpax.unpack(packed, unquote(opts))
```
Comment (whatyouhide) [inline test/test_helper.exs]: Why did you remove these assertions?

**Comment by lexmag** [comment]
Hi Hidetaka, thanks for your contribution.

I think we could name this option differently (`:fields`, `:keys`, other name?) not to make confusion that its opposite option exists, e.g. `:skip`, `:except`.

I like to not have opposite option as it's not clear if `__struct__` will be included or not, and if someone really wants all the keys from the struct – `Map.keys(%MyStruct{})` can be used for that.

@whatyouhide @HidetakaKojo what are your thoughts on this?

**Comment by whatyouhide** [comment]
I like `:fields` instead of `:only`, it even conveys the meaning of what's happening better IMO. I also agree that it's better not to have a "skip" option because indeed it's confusing with `__struct__`.

It's a shame this wasn't discussed in Poison first because it would've been neat to call this option the same :\

File: test/test_helper.exs (hunk 2 above)
Comment (HidetakaKojo) [inline test/test_helper.exs]: In my environment, failed to compile.
```
bash-3.2$ elixir --version
Erlang/OTP 18 [erts-7.1] [source] [64-bit] [smp:4:4] [async-threads:10] [hipe] [kernel-poll:false] [dtrace]

Elixir 1.2.0
bash-3.2$ mix test
** (CompileError) test/msgpax/ext_test.exs:39: undefined function packed/0
    (stdlib) lists.erl:1337: :lists.foreach/2
    (stdlib) erl_eval.erl:669: :erl_eval.do_apply/6
```

But, Successful after updating.
```
bash-3.2$ elixir --version
Erlang/OTP 18 [erts-7.1] [source] [64-bit] [smp:4:4] [async-threads:10] [hipe] [kernel-poll:false] [dtrace]

Elixir 1.2.3
bash-3.2$ mix test
...........................................................

Finished in 1.8 seconds (0.9s on load, 0.8s on tests)
59 tests, 0 failures

Randomized with seed 837795
```

**Comment by HidetakaKojo** [comment]
Yes, I think so. :+1:

**Comment by lexmag** [comment]
@HidetakaKojo, the patch looks wonderful, please squash everything into single commit and I'll merge it right after that.

**Comment by HidetakaKojo** [comment]
Thanks, @lexmag.

**Comment by lexmag** [comment]
:yellow_heart:

**Comment by whatyouhide** [comment]
Thank *you* @HidetakaKojo :clap: =
